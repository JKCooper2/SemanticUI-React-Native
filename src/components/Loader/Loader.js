// @flow
import React, { PureComponent } from 'react';
import { View, Text, ActivityIndicator } from 'react-native';

import type { LoaderProps } from '~/types/components/Loader';

class Loader extends PureComponent<LoaderProps> {
    static defaultProps = {
        active: false,
        inverted: false
    }

    render() {
        let { children, content, active, inverted } = this.props;

        return (
            <View style={{ justifyContent: 'center', alignItems: 'center' }}>
                <ActivityIndicator
                    animating={active}
                    color={inverted ? 'white' : 'black'}
                />
                {(!!children || !!content) && (
                    <Text style={{ color: inverted ? 'white' : 'black' }}>{children || content}</Text>
                )}
            </View>
        );
    }
}

export default Loader;