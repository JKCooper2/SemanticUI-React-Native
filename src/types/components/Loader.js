
export type LoaderProps = {
    children?: string;
    content?: string;
    active?: boolean;
    inverted?: boolean;
}