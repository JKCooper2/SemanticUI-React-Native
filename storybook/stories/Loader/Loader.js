// @flow
import React, { Fragment } from 'react';
import { View } from 'react-native';

import { storiesOf } from '@storybook/react-native';

import Loader from '~/components/Loader';

const stories = storiesOf('Loader', module);

stories.add('Standard', () => (
    <Fragment>
        <Loader active />
        <View style={{ backgroundColor: 'black' }}>
            <Loader active inverted />
            <Loader active inverted>
                Loading
            </Loader>
            <Loader active inverted content="Loading Content"/>
        </View>
    </Fragment>
));

export default stories;
