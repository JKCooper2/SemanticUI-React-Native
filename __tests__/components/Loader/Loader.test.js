// @flow
import React from 'react';
import { shallow } from 'enzyme';

import Loader from '~/components/Loader';

test('renders correctly', () => {
    const loader = shallow(<Loader active />);
    expect(loader).toMatchSnapshot();
});
