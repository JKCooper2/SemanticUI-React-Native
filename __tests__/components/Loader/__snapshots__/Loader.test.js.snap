// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Component
    animating={true}
    color="black"
    hidesWhenStopped={true}
    size="small"
  />
</Component>
`;
